controllers/storagecluster/reconcile.go

```diff
@@ -723,3 +734,57 @@ func validateOverprovisionControlSpec(sc *ocsv1.StorageCluster, reqLogger logr.L
 	}
 	return nil
 }
+
+func validateCustomStorageClassNames(sc *ocsv1.StorageCluster, reqLogger logr.Logger) error {
+	var err error
+	scMap := make(map[string]bool)
+	if sc.Spec.ManagedResources.CephBlockPools.StorageClassName != "" {
+		if _, has := scMap[sc.Spec.ManagedResources.CephBlockPools.StorageClassName]; has {
+			err = fmt.Errorf("Duplicate storage class name provided for CephBlockPools")
```

Comment:
This error will be overridden by errors from other lines. We need to combine all errors and return it in the end.
And to reduce the amount of logs, let's not log it after every check. Let's log it once in the end.

Reply:
Use an array and append the errors to it and log all of them together?

Reply:
Yes. Can use errors.Join (needs go 1.20) to merge errors.

Reply:
Ok, updated the code to use an array for concatenating all the errors and logging in the end of the function, as the project still uses go 1.19